internal/state/statuses.go

```diff
@@ -109,17 +113,25 @@ func buildStatuses(graph *graph) Statuses {
 		parentStatuses := make(map[string]ParentStatus)
 
 		for ref := range r.ValidSectionNameRefs {
-			parentStatuses[ref] = ParentStatus{
-				Attached: gcValidAndExist, // Attached only when GatewayClass is valid and exists
-			}
+			parentStatuses[ref] = ParentStatus{}
 		}
-		for ref := range r.InvalidSectionNameRefs {
+		for ref, cond := range r.InvalidSectionNameRefs {
 			parentStatuses[ref] = ParentStatus{
-				Attached: false,
+				Conditions: []conditions.RouteCondition{cond},
```

Comment:
I refactored it, so that each parent ref includes all conditions

I also moved deduplication logic to the `conditions` package. Right now, the logic is simple - use the last one. However, it makes sense to me to keep it there, so that the knowledge how to combine conditions is in that package too.

The refactoring adds some complexity to the unit tests in `state` package, but at the same time it also simplifies `status.prepareHTTPRouteStatus`

I'm not worrying about some duplicates across  all parentRefs - because of the deduplication, the amount of possible conditions is small, and I don't expect the number of parentRefs to be big.